tools/echo_server/echo_server.go

```diff
@@ -189,5 +242,22 @@ func main() {
 		logSetup()
 	}
 
-	startup(config)
+	// overwrite config only if the flags are set
+	flag.Visit(func (f *flag.Flag) {
+		if f.Name == portFlagName {
+			config.ServerPort = *serverPort
+		} else if f.Name == repeatFlagName {
+			config.RepeatMode = *repeatMode
+		} else if f.Name == intervalFlagName {
+			config.RepeatIntervalSeconds = *repeatIntervalSeconds
+		} else if f.Name == certFlagName {
+			config.ServerCert = *serverCert
+			config.Secure = true
+		} else if f.Name == keyFlagName {
```

Comment:
If secure is enabled when specifying the key or server cert, we may as well just remove the secure option. Also should probably check that both are defined.

Reply:
I want to leave the secure option for users who want to use config file instead, it will be easier for them to toggle secure mode without making too many changes to the config file.

I agree that we should enable secure mode ONLY WHEN both cmdline args are present, will make this change in next rev.